flow/project.py

```diff
@@ -636,14 +1082,25 @@ def hello(job):
         from the environment.
     :type config:
         A signac config object.
+    :param entrypoint:
+        A dictionary with two possible keys: executable and path. Path
+        represents the filepath location of the script file (the script file
+        must call `main`). Executable represents the location to the python
+        interpreter used for the executable of `FlowOperation` that are Python
+        functions.
+    :type entrypoint:
+        dict
     """
 
-    def __init__(self, config=None, environment=None):
+    def __init__(self, config=None, environment=None, entrypoint=None):
```

Comment:
In python, `dict` and `list` are mutable and should not be used for default parameters. This can lead to the default values being modified (e.g. key, value pairs being added to `entrypoint`). `None` is safer.